### ISSUE TITLE
Use actual color passed to set_sky() for night sky

### ISSUE THREAD

**Issue by LoneWolfHT** [minetest/minetest#9508]
## Problem

When I set the night_sky color to pure white (night_sky = 0xffffff inside of player:set_sky) I get this:
![image](https://user-images.githubusercontent.com/28972859/76589176-721be180-64a6-11ea-937d-9889560234f6.png)

## Solutions

It'd be nice if the darkening could be removed or made toggle-able, I want full control over the color of my night sky

**Comment by paramat** [comment]
Seems a good idea, in some form.

**Comment by TheTermos** [comment]
Could be done I guess, instead of being multiplied by brightness it could just gradually blend into night colors during sunset and into day colors during sunrise.

Now is the best time before mods start making use of improved set_sky.

**Comment by Wuzzy2** [comment]
Yes, please. This color offset is confusing and also undocumented, so I'm basically forced to guess the result. >_>

**Comment by Wuzzy2** [comment]
Can this be fixed before release, please?

If you ignore this for 5.2.0, you will have less work for this release, obviously, but it will bite you in the ass later, when it's finally fixed because you then have to figure out annoying backwards-compability stuff, resulting in a more ugly API.
If you decide fix it for 5.2.0, you will release with a clean, future-proof API to begin with. But it's more work.

Your choice. :-)

Anyway, if you want to religiously stick to the feature freeze, one alternative is to *at least* document the current (weird) behavior. (The fact that night sky is made much darker than you expect feels like a bug, tbh.)

**Comment by LoneWolfHT** [comment]
I'll take a peek tomorrow, hopefully I'll beak able to make head and tails of it

**Comment by LoneWolfHT** [comment]
Well, you can remove the darkening but node darkness at night won't respect that. So you could have a pure white sky and the terrain would still be pretty much pure black.
There are also quite a few other things that depend on the sky darkening. This won't be a simple fix and may be over my head

**Comment by Wuzzy2** [comment]
> Well, you can remove the darkening but node darkness at night won't respect that.

I think this is OK. It's called `set_sky`, not `set_sky_and_also_other_unrelated_node_shading_stuff_because_we_like_to_blow_up_our_api_calls`.

**Comment by LoneWolfHT** [comment]
Hmmm, it looks like the node/sky darkness can be set with override_day_night_ratio().
Shall I just remove the sky/cloud darkening and leave it at that?

**Comment by paramat** [comment]
> Can this be fixed before release, please?

It is a feature not a bug.
I understand why it would be good to do this now before set_sky is used, but altering the colouration of night sky is not trivial and is likely to throw up unforseen bugs and difficulties. Releasing 5.2.0 is already difficult enough.

So i suggest set_sky is documented as 'unstable in terms of night sky colouring' for 5.2.0, it will not be much disruption for modders to change their mods later. Besides, 5.3.0 is planned to be soon.

**Comment by paramat** [comment]
> Hmmm, it looks like the node/sky darkness can be set with override_day_night_ratio().
Shall I just remove the sky/cloud darkening and leave it at that?

Override day night ratio is not trivial to use and will cause difficulties.

Please, lets not rush something in that is highly likely to cause problems, ater all the work to get new set_sky merged. Lets leave this until after release and just add a comment about instability to documentation.

**Comment by v-rob** [comment]
Perhaps we could mark the whole `set_sky` and co. API as unstable until 5.3.0 because it's such a large thing and bugs have been found already.  That way, other unforeseen bad features/bugs besides this one won't disrupt anything because it's all considered unstable.

**Comment by TheTermos** [comment]
Here's a quick proof of concept:
https://github.com/TheTermos/minetest/tree/sky-blend

I didn't bother with indoor and plain sky yet so they might work incorrectly. Night sky colors are random too, so just play around with set_sky().
Dawn sky color isn't used.

**Comment by paramat** [comment]
> Perhaps we could mark the whole set_sky and co. API as unstable until 5.3.0

That might be best.

**Comment by Wuzzy2** [comment]
@paramat: But currently, the current 5.1.1-style `set_sky` syntax has been marked as “deprecated”. It's not a good idea to deprecate an API only to replace it with an experimental one ...
When you mark the new `set_sky` API as experimental, please don't forget to UNmark the current stable `set_sky` as deprecated.

**Comment by v-rob** [comment]
Unmark as deprecated, but also give notice that it will be deprecated in the future and replaced with the new stuff.

**Comment by paramat** [comment]
It just occurred to me that dawn sky colours might need the same change, as those are probably darkened like night sky colours are.

**Comment by paramat** [comment]
Confirmed, so this issue applies to dawn colours too.
Screenshot is with dawn colours set to pure white:

![screenshot_20200402_164543](https://user-images.githubusercontent.com/3686677/78277065-ccd7b480-750b-11ea-887f-7c8ad46b0b09.png)

**Comment by TheTermos** [comment]
I find dawn sky redundant, without it things are simple, night blends into day and vice versa. Including additional color in the cycle makes it considerably more complex for no real benefit. Also if there's separate dawn sky, why there's no separate dusk sky? That seems another random unfinished idea from the old days.

**Comment by LoneWolfHT** [comment]
I don't mind it, it adds a nice touch. I also think the dawn sky is used for dusk too

**Comment by paramat** [comment]
> for no real benefit

There is benefit for players, many find the dawn/dusk stages beautiful.
I could agree with you considering this in isolation, but the dawn/dusk stages are part of Minetest's history and character and are liked by very many, so cannot be casually discarded. Those who want to radically change many aspects of the sky should use the new customisation features to do so.
> the dawn sky is used for dusk too

It is. As is the 'sunrise' texture too.
The usage of the words 'dawn' and 'sunrise' (for the texture) are sloppy and misleading (hmm, maybe that is another necessary change needed to the new set_sky API?).

**Comment by TheTermos** [comment]
> There is benefit for players, many find the dawn/dusk stages beautiful.

Obviously, we're not talking about getting rid of the stages, only about separate dawn color definition.
Check out the poc I linked above, dawn/dusk stages are definitely still there, even though dawn sky color isn't used.

- If this suggestion is going to be implemented then dawn sky color has to be changed
because it wouldn't be affected by world brightness anymore.
- if it is to be changed, then why shouldn't it be a blend between day & night? That definitely makes sense.
- if it's a blend then it can be calculated, no need for separate definition.

Anyway, I could code the PR if it involves getting rid of the third color, otherwise maybe someone else will.

**Comment by Wuzzy2** [comment]
Why not keeping the dawn color but making it optional? The default could be a calculated value.
